### ISSUE TITLE
`override_repo` does not suggest using repo name for module override

### ISSUE THREAD

**Issue by luispadron** [bazelbuild/bazel#23981]
### Description of the bug:

I have a dependency on a **Bazel module** [swift-syntax](https://registry.bazel.build/modules/swift-syntax), for reasons we override this module using `archive_override`:

```starlark
SWIFT_SYNTAX_VERSION = "510.0.3"

bazel_dep(
    name = "swift-syntax",
    version = SWIFT_SYNTAX_VERSION,
    repo_name = "SwiftSyntax",
)

archive_override(
    module_name = "swift-syntax",
    integrity = "sha256-8HShak9S6SPqhVyHqV89Jnz37KdNa907iyPnSuWfpys=",
    strip_prefix = "swift-syntax-%s" % SWIFT_SYNTAX_VERSION,
    urls = [
        "https://github.com/square/swift-syntax-prebuilt/releases/download/{version}/swift-syntax-{version}.tar.gz".format(
            version = SWIFT_SYNTAX_VERSION,
        ),
    ],
)
```

I use another **Bazel module** [rules_swift_package_manager](https://registry.bazel.build/modules/rules_swift_package_manager) which provides a module extension `swift_deps` that provides **Bazel repositories** for use within the build. This `swift_deps` extension proxy generates a repository named `swiftpkg_swift_syntax`.

```starlark
swift_deps = use_extension(
    "@rules_swift_package_manager//:extensions.bzl",
    "swift_deps",
)
swift_deps.from_package(
    declare_swift_deps_info = True,
    resolved = "Package.resolved",
    swift = "Package.swift",
)
use_repo(
    swift_deps,
    "swift_deps_info",
    "swiftpkg_swift_syntax",
)
```

I'd like to replace the `swiftpkg_swift_syntax` repository with my override `swift-syntax` repository, I use the new `override_repo` feature in Bazel 7.4+:

```starlark
override_repo(
    swift_deps,
    swiftpkg_swift_syntax = "swift-syntax",
)
```

I get the following error:

```sh
bazel build //...

Starting local Bazel server and connecting to it...
ERROR: /Users/lpadron/Desktop/override_repo_example/MODULE.bazel:87:14: Traceback (most recent call last):
        File "/Users/lpadron/Desktop/override_repo_example/MODULE.bazel", line 87, column 14, in <toplevel>
                override_repo(
Error in override_repo: The repo exported as 'swiftpkg_swift_syntax' by module extension 'swift_deps' is overridden with 'swift-syntax', but no repo is visible under this name
ERROR: Error computing the main repository mapping: error executing MODULE.bazel file for the root module
```

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

`bazel build //...` on the attached zip.

[override_repo_example.zip](https://github.com/user-attachments/files/17380922/override_repo_example.zip)

### Which operating system are you running Bazel on?

macOS 14.6.1

### What is the output of `bazel info release`?

```
release 7.4.0rc1
```

**Comment by fmeum** [comment]
Could you try this?
```starlark
override_repo(
    swift_deps,
    swiftpkg_swift_syntax = "SwiftSyntax",
)
```

You need to specify apparent repo names, not module names.

**Comment by luispadron** [comment]
Oh, that does work and it overrides the repository as expected. Do you know where `SwiftSyntax` is coming from as the repo name?

At a minimum maybe the error message could suggest this if that seems doable

**Comment by luispadron** [comment]
@fmeum Im getting a suggestion now but its suggesting the thing im trying to override (it should suggest the `SwiftSyntax` repo instead):


```
Error in override_repo: The repo exported as 'swiftpkg_swift_syntax' by module extension 'swift_deps' is overridden with 'swift-syntax', but no repo is visible under this name (did you mean 'swiftpkg_swift_syntax'?)
ERROR: Error computing the main repository mapping: error executing MODULE.bazel file for the root module
```

**Comment by fmeum** [comment]
@luispadron Good catch, should be fixed.

**Comment by luispadron** [comment]
Nice!

```
Error in override_repo: The repo exported as 'swiftpkg_swift_syntax' by module extension 'swift_deps' is overridden with 'swift-syntax', but no repo is visible under this name (did you mean 'SwiftSyntax'?)
```

**Comment by luispadron** [comment]
Thanks for your help here!

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC3](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc3). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc3`. Thanks!
